pkg/controller/manage/kmesh_manage.go

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kmeshmanage
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"syscall"
+	"time"
+
+	netns "github.com/containernetworking/plugins/pkg/ns"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	ns "kmesh.net/kmesh/pkg/controller/netns"
+	"kmesh.net/kmesh/pkg/logger"
+)
+
+var (
+	log                = logger.NewLoggerField("kmesh_manage")
+	annotationDelPatch = []byte(fmt.Sprintf(
+		`{"metadata":{"annotations":{"%s":null}}}`,
+		KmeshAnnotation,
+	))
+	annotationAddPatch = []byte(fmt.Sprintf(
+		`{"metadata":{"annotations":{"%s":"%s"}}}`,
+		KmeshAnnotation,
+		"enabled",
+	))
+)
+
+const (
+	DefaultInformerSyncPeriod = 30 * time.Second
+	LabelSelectorKmesh        = "istio.io/dataplane-mode=Kmesh"
+	KmeshAnnotation           = "kmesh.net/redirection"
+)
+
+func StartKmeshManageController(client kubernetes.Interface) error {
+	stopChan := make(chan struct{})
+	nodeName := os.Getenv("NODE_NAME")
+
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(client, DefaultInformerSyncPeriod,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.FieldSelector = fmt.Sprintf("spec.nodeName=%s", nodeName)
+			options.LabelSelector = LabelSelectorKmesh
+		}))
+
+	informerFactory.Start(wait.NeverStop)
+	informerFactory.WaitForCacheSync(wait.NeverStop)
+
+	podInformer := informerFactory.Core().V1().Pods().Informer()
+
+	if _, err := podInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			pod, ok := obj.(*corev1.Pod)
+			if !ok {
+				log.Errorf("expected *corev1.Pod but got %T", obj)
+				return
+			}
+			namespace, err := client.CoreV1().Namespaces().Get(context.TODO(), pod.GetNamespace(), metav1.GetOptions{})
```

Comment:
This logic of checking if ns has a label can be abstracted into a common function； used by AddFunc and  DeleteFunc

Reply:
> This logic of checking if ns has a label can be abstracted into a common function； used by AddFunc and DeleteFunc

I have deleted this logic, because it may affect the data in bpf map kmesh_manage when kmesh restart